src/main/java/database/service/service/JsonDatabaseService.java

```diff
@@ -0,0 +1,195 @@
+        jsonDatabaseFile.createNewFile();
+        Files.writeString(jsonDatabaseFile.toPath(), "[]");
+        return Files.exists(jsonDatabaseFile.toPath());
+    }
+
+    @Override
+    public boolean deleteTable(Class<? extends BaseEntity> entityClass) throws IOException {
+        Path databasePath = Path.of(getDatabasePath(entityClass));
+        Files.delete(databasePath);
+        return !Files.exists(databasePath);
+    }
+
+    @Override
+    public <T extends BaseEntity> T addNewRecordToTable(T entity) throws DatabaseDoesNotExistException, IOException {
+        Class<? extends BaseEntity> entityClass = entity.getClass();
+        Path databasePath = Path.of(getDatabasePath(entityClass));
+
+        if (!Files.exists(databasePath)) {
+            throw new DatabaseDoesNotExistException("Database does not exist.");
+        }
+
+        List<T> entities = deserializeEntities(entityClass, Files.readString(databasePath));
+        String entityClassName = entityClass.getName();
+        var lastUsedId = entityIds.get(entityClassName);
+
+        if (lastUsedId == null) {
+            if (entities.isEmpty()) {
+                entityIds.put(entityClassName, -1);
+            } else {
+                int lastEntityId = entities.get(entities.size() - 1).getId();
+                entityIds.put(entityClassName, lastEntityId);
+            }
+        }
+
+        entity.setId(entityIds.merge(entityClassName, 1, Integer::sum));
+        entities.add(entity);
+
+        objectMapper.writeValue(databasePath.toFile(), entities);
+        return entity;
+    }
+
+    @Override
+    public <T extends BaseEntity, I> T updateRecordInTable(T entity, I id) throws IOException,
+            ProvidedIdDoesNotExistException {
+        Class<? extends BaseEntity> entityClass = entity.getClass();
+        Path databasePath = Path.of(getDatabasePath(entityClass));
+
+        List<T> entities = deserializeEntities(entityClass, Files.readString(databasePath));
+
+        Optional<T> optionalEntity = entities.stream().filter(e -> id.equals(e.getId())).findFirst();
+        T entityFoundById;
+        if (optionalEntity.isPresent()) {
+            entityFoundById = optionalEntity.get();
+            updateEntityFields(entityFoundById, entity);
+        } else {
+            throw new ProvidedIdDoesNotExistException("Entity with provided Id does not exist.");
+        }
+
+        objectMapper.writeValue(databasePath.toFile(), entities);
+        return entityFoundById;
+    }
+
+    @Override
+    public <I> boolean removeRecordFromTable(Class<? extends BaseEntity> entityClass, I id) throws IOException {
+        Path databasePath = Path.of(getDatabasePath(entityClass));
+
+        List<? extends BaseEntity> entities = deserializeEntities(entityClass, Files.readString(databasePath));
+
+        boolean isEntityRemoved = entities.removeIf(e -> id.equals(e.getId()));
+
+        objectMapper.writeValue(databasePath.toFile(), entities);
```

Comment:
Please call `objectMapper.writeValue` only if entity was removed from the list, otherwise no need to do this.